Week2/6.5.group_by.py

```diff
@@ -0,0 +1,16 @@
+def group_by(func, iterable):
+    """
+    Group the elements of an iterable by the value returned by the given function.
+    :param func: The function to be used to group the elements
+    :param iterable: The iterable that pass to func
+    :return:
+    """
+    result = {}
+    for item in iterable:
```

Comment:
nice solution, but can you make it in less lines? you can use list comprehension and dict methods that simplify your if statement.